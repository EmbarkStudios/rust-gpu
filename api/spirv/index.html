<!DOCTYPE html><html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1.0"><meta name="generator" content="rustdoc"><meta name="description" content="The SPIR-V header."><meta name="keywords" content="rust, rustlang, rust-lang, spirv"><title>spirv - Rust</title><link rel="preload" as="font" type="font/woff2" crossorigin href="../SourceSerif4-Regular.ttf.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../FiraSans-Regular.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../FiraSans-Medium.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../SourceCodePro-Regular.ttf.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../SourceSerif4-Bold.ttf.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../SourceCodePro-Semibold.ttf.woff2"><link rel="stylesheet" type="text/css" href="../normalize.css"><link rel="stylesheet" type="text/css" href="../rustdoc.css" id="mainThemeStyle"><link rel="stylesheet" type="text/css" href="../ayu.css" disabled><link rel="stylesheet" type="text/css" href="../dark.css" disabled><link rel="stylesheet" type="text/css" href="../light.css" id="themeStyle"><script id="default-settings" ></script><script src="../storage.js"></script><script src="../crates.js"></script><script defer src="../main.js"></script>
    <noscript><link rel="stylesheet" href="../noscript.css"></noscript><link rel="alternate icon" type="image/png" href="../favicon-16x16.png"><link rel="alternate icon" type="image/png" href="../favicon-32x32.png"><link rel="icon" type="image/svg+xml" href="../favicon.svg"></head><body class="rustdoc mod crate"><!--[if lte IE 11]><div class="warning">This old browser is unsupported and will most likely display funky things.</div><![endif]--><nav class="mobile-topbar"><button class="sidebar-menu-toggle">&#9776;</button><a class="sidebar-logo" href="../spirv/index.html"><div class="logo-container"><img class="rust-logo" src="../rust-logo.svg" alt="logo"></div>
        </a><h2 class="location"></h2>
    </nav>
    <nav class="sidebar"><a class="sidebar-logo" href="../spirv/index.html"><div class="logo-container"><img class="rust-logo" src="../rust-logo.svg" alt="logo"></div>
        </a><h2 class="location"><a href="#">Crate spirv</a></h2><div class="sidebar-elems"><div class="block"><ul><li class="version">Version 0.2.0+1.5.4</li><li><a id="all-types" href="all.html">All Items</a></li></ul></div><section><div class="block"><ul><li><a href="#structs">Structs</a></li><li><a href="#enums">Enums</a></li><li><a href="#constants">Constants</a></li><li><a href="#types">Type Definitions</a></li></ul></div></section><div id="sidebar-vars" data-name="spirv" data-ty="mod" data-relpath=""></div><script defer src="sidebar-items.js"></script></div></nav><main><div class="width-limiter"><div class="sub-container"><a class="sub-logo-container" href="../spirv/index.html"><img class="rust-logo" src="../rust-logo.svg" alt="logo"></a><nav class="sub"><form class="search-form"><div class="search-container"><span></span><input class="search-input" name="search" autocomplete="off" spellcheck="false" placeholder="Click or press ‘S’ to search, ‘?’ for more options…" type="search"><button type="button" id="help-button" title="help">?</button><div id="settings-menu" tabindex="-1">
                                <a href="../settings.html" title="settings"><img width="22" height="22" alt="Change settings" src="../wheel.svg"></a></div>
                        </div></form></nav></div><section id="main-content" class="content"><div class="main-heading">
    <h1 class="fqn"><span class="in-band">Crate <a class="mod" href="#">spirv</a><button id="copy-path" onclick="copy_path(this)" title="Copy item path to clipboard"><img src="../clipboard.svg" width="19" height="18" alt="Copy item path"></button></span></h1><span class="out-of-band"><a class="srclink" href="../src/spirv/lib.rs.html#1-14">source</a> · <a id="toggle-all-docs" href="javascript:void(0)" title="collapse all docs">[<span class="inner">&#x2212;</span>]</a></span></div><details class="rustdoc-toggle top-doc" open><summary class="hideme"><span>Expand description</span></summary><div class="docblock"><p>The SPIR-V header.</p>
<p>This crate contains Rust definitions of all SPIR-V structs, enums,
and constants.</p>
<p>The version of this crate is the version of SPIR-V it contains.</p>
</div></details><h2 id="structs" class="small-section-header"><a href="#structs">Structs</a></h2>
<div class="item-table"><div class="item-row"><div class="item-left module-item"><a class="struct" href="struct.FPFastMathMode.html" title="spirv::FPFastMathMode struct">FPFastMathMode</a></div><div class="item-right docblock-short"><p>SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_fp_fast_math_mode_a_fp_fast_math_mode">FPFastMathMode</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="struct" href="struct.FragmentShadingRate.html" title="spirv::FragmentShadingRate struct">FragmentShadingRate</a></div><div class="item-right docblock-short"><p>SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_fragment_shading_rate_a_fragment_shading_rate">FragmentShadingRate</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="struct" href="struct.FunctionControl.html" title="spirv::FunctionControl struct">FunctionControl</a></div><div class="item-right docblock-short"><p>SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_function_control_a_function_control">FunctionControl</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="struct" href="struct.ImageOperands.html" title="spirv::ImageOperands struct">ImageOperands</a></div><div class="item-right docblock-short"><p>SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_image_operands_a_image_operands">ImageOperands</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="struct" href="struct.KernelProfilingInfo.html" title="spirv::KernelProfilingInfo struct">KernelProfilingInfo</a></div><div class="item-right docblock-short"><p>SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_kernel_profiling_info_a_kernel_profiling_info">KernelProfilingInfo</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="struct" href="struct.LoopControl.html" title="spirv::LoopControl struct">LoopControl</a></div><div class="item-right docblock-short"><p>SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_loop_control_a_loop_control">LoopControl</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="struct" href="struct.MemoryAccess.html" title="spirv::MemoryAccess struct">MemoryAccess</a></div><div class="item-right docblock-short"><p>SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_memory_access_a_memory_access">MemoryAccess</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="struct" href="struct.MemorySemantics.html" title="spirv::MemorySemantics struct">MemorySemantics</a></div><div class="item-right docblock-short"><p>SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_memory_semantics_a_memory_semantics">MemorySemantics</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="struct" href="struct.RayFlags.html" title="spirv::RayFlags struct">RayFlags</a></div><div class="item-right docblock-short"><p>SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_ray_flags_a_ray_flags">RayFlags</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="struct" href="struct.SelectionControl.html" title="spirv::SelectionControl struct">SelectionControl</a></div><div class="item-right docblock-short"><p>SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_selection_control_a_selection_control">SelectionControl</a></p>
</div></div></div><h2 id="enums" class="small-section-header"><a href="#enums">Enums</a></h2>
<div class="item-table"><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.AccessQualifier.html" title="spirv::AccessQualifier enum">AccessQualifier</a></div><div class="item-right docblock-short"><p>/// SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_access_qualifier_a_access_qualifier">AccessQualifier</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.AddressingModel.html" title="spirv::AddressingModel enum">AddressingModel</a></div><div class="item-right docblock-short"><p>/// SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_addressing_model_a_addressing_model">AddressingModel</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.BuiltIn.html" title="spirv::BuiltIn enum">BuiltIn</a></div><div class="item-right docblock-short"><p>/// SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_built_in_a_built_in">BuiltIn</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.CLOp.html" title="spirv::CLOp enum">CLOp</a></div><div class="item-right docblock-short"><p><a href="https://www.khronos.org/registry/spir-v/specs/unified1/OpenCL.ExtendedInstructionSet.100.html">OpenCL.std</a> extended instruction opcode</p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.Capability.html" title="spirv::Capability enum">Capability</a></div><div class="item-right docblock-short"><p>/// SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_capability_a_capability">Capability</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.Decoration.html" title="spirv::Decoration enum">Decoration</a></div><div class="item-right docblock-short"><p>/// SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_decoration_a_decoration">Decoration</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.Dim.html" title="spirv::Dim enum">Dim</a></div><div class="item-right docblock-short"><p>/// SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_dim_a_dim">Dim</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.ExecutionMode.html" title="spirv::ExecutionMode enum">ExecutionMode</a></div><div class="item-right docblock-short"><p>/// SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_execution_mode_a_execution_mode">ExecutionMode</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.ExecutionModel.html" title="spirv::ExecutionModel enum">ExecutionModel</a></div><div class="item-right docblock-short"><p>/// SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_execution_model_a_execution_model">ExecutionModel</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.FPRoundingMode.html" title="spirv::FPRoundingMode enum">FPRoundingMode</a></div><div class="item-right docblock-short"><p>/// SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_fp_rounding_mode_a_fp_rounding_mode">FPRoundingMode</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.FunctionParameterAttribute.html" title="spirv::FunctionParameterAttribute enum">FunctionParameterAttribute</a></div><div class="item-right docblock-short"><p>/// SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_function_parameter_attribute_a_function_parameter_attribute">FunctionParameterAttribute</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.GLOp.html" title="spirv::GLOp enum">GLOp</a></div><div class="item-right docblock-short"><p><a href="https://www.khronos.org/registry/spir-v/specs/unified1/GLSL.std.450.html">GLSL.std.450</a> extended instruction opcode</p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.GroupOperation.html" title="spirv::GroupOperation enum">GroupOperation</a></div><div class="item-right docblock-short"><p>/// SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_group_operation_a_group_operation">GroupOperation</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.ImageChannelDataType.html" title="spirv::ImageChannelDataType enum">ImageChannelDataType</a></div><div class="item-right docblock-short"><p>/// SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_image_channel_data_type_a_image_channel_data_type">ImageChannelDataType</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.ImageChannelOrder.html" title="spirv::ImageChannelOrder enum">ImageChannelOrder</a></div><div class="item-right docblock-short"><p>/// SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_image_channel_order_a_image_channel_order">ImageChannelOrder</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.ImageFormat.html" title="spirv::ImageFormat enum">ImageFormat</a></div><div class="item-right docblock-short"><p>/// SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_image_format_a_image_format">ImageFormat</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.KernelEnqueueFlags.html" title="spirv::KernelEnqueueFlags enum">KernelEnqueueFlags</a></div><div class="item-right docblock-short"><p>/// SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_kernel_enqueue_flags_a_kernel_enqueue_flags">KernelEnqueueFlags</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.LinkageType.html" title="spirv::LinkageType enum">LinkageType</a></div><div class="item-right docblock-short"><p>/// SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_linkage_type_a_linkage_type">LinkageType</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.MemoryModel.html" title="spirv::MemoryModel enum">MemoryModel</a></div><div class="item-right docblock-short"><p>/// SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_memory_model_a_memory_model">MemoryModel</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.Op.html" title="spirv::Op enum">Op</a></div><div class="item-right docblock-short"><p>SPIR-V <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_instructions_a_instructions">instructions</a> opcodes</p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.RayQueryCandidateIntersectionType.html" title="spirv::RayQueryCandidateIntersectionType enum">RayQueryCandidateIntersectionType</a></div><div class="item-right docblock-short"><p>/// SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_ray_query_candidate_intersection_type_a_ray_query_candidate_intersection_type">RayQueryCandidateIntersectionType</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.RayQueryCommittedIntersectionType.html" title="spirv::RayQueryCommittedIntersectionType enum">RayQueryCommittedIntersectionType</a></div><div class="item-right docblock-short"><p>/// SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_ray_query_committed_intersection_type_a_ray_query_committed_intersection_type">RayQueryCommittedIntersectionType</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.RayQueryIntersection.html" title="spirv::RayQueryIntersection enum">RayQueryIntersection</a></div><div class="item-right docblock-short"><p>/// SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_ray_query_intersection_a_ray_query_intersection">RayQueryIntersection</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.SamplerAddressingMode.html" title="spirv::SamplerAddressingMode enum">SamplerAddressingMode</a></div><div class="item-right docblock-short"><p>/// SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_sampler_addressing_mode_a_sampler_addressing_mode">SamplerAddressingMode</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.SamplerFilterMode.html" title="spirv::SamplerFilterMode enum">SamplerFilterMode</a></div><div class="item-right docblock-short"><p>/// SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_sampler_filter_mode_a_sampler_filter_mode">SamplerFilterMode</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.Scope.html" title="spirv::Scope enum">Scope</a></div><div class="item-right docblock-short"><p>/// SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_scope_a_scope">Scope</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.SourceLanguage.html" title="spirv::SourceLanguage enum">SourceLanguage</a></div><div class="item-right docblock-short"><p>/// SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_source_language_a_source_language">SourceLanguage</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.StorageClass.html" title="spirv::StorageClass enum">StorageClass</a></div><div class="item-right docblock-short"><p>/// SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_storage_class_a_storage_class">StorageClass</a></p>
</div></div></div><h2 id="constants" class="small-section-header"><a href="#constants">Constants</a></h2>
<div class="item-table"><div class="item-row"><div class="item-left module-item"><a class="constant" href="constant.MAGIC_NUMBER.html" title="spirv::MAGIC_NUMBER constant">MAGIC_NUMBER</a></div><div class="item-right docblock-short"></div></div><div class="item-row"><div class="item-left module-item"><a class="constant" href="constant.MAJOR_VERSION.html" title="spirv::MAJOR_VERSION constant">MAJOR_VERSION</a></div><div class="item-right docblock-short"></div></div><div class="item-row"><div class="item-left module-item"><a class="constant" href="constant.MINOR_VERSION.html" title="spirv::MINOR_VERSION constant">MINOR_VERSION</a></div><div class="item-right docblock-short"></div></div><div class="item-row"><div class="item-left module-item"><a class="constant" href="constant.REVISION.html" title="spirv::REVISION constant">REVISION</a></div><div class="item-right docblock-short"></div></div></div><h2 id="types" class="small-section-header"><a href="#types">Type Definitions</a></h2>
<div class="item-table"><div class="item-row"><div class="item-left module-item"><a class="type" href="type.Word.html" title="spirv::Word type">Word</a></div><div class="item-right docblock-short"></div></div></div></section></div></main><div id="rustdoc-vars" data-root-path="../" data-current-crate="spirv" data-themes="ayu,dark,light" data-resource-suffix="" data-rustdoc-version="1.63.0-nightly (546c826f0 2022-06-12)" ></div>
</body></html>