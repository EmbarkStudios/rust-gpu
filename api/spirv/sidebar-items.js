window.SIDEBAR_ITEMS = {"constant":["MAGIC_NUMBER","MAJOR_VERSION","MINOR_VERSION","REVISION"],"enum":["AccessQualifier","AddressingModel","BuiltIn","CLOp","Capability","Decoration","Dim","ExecutionMode","ExecutionModel","FPRoundingMode","FunctionParameterAttribute","GLOp","GroupOperation","ImageChannelDataType","ImageChannelOrder","ImageFormat","KernelEnqueueFlags","LinkageType","MemoryModel","Op","RayQueryCandidateIntersectionType","RayQueryCommittedIntersectionType","RayQueryIntersection","SamplerAddressingMode","SamplerFilterMode","Scope","SourceLanguage","StorageClass"],"struct":["FPFastMathMode","FragmentShadingRate","FunctionControl","ImageOperands","KernelProfilingInfo","LoopControl","MemoryAccess","MemorySemantics","RayFlags","SelectionControl"],"type":["Word"]};