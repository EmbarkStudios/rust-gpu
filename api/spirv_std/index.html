<!DOCTYPE html><html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1.0"><meta name="generator" content="rustdoc"><meta name="description" content="Core functions, traits, and more that make up a “standard library” for SPIR-V for use in rust-gpu."><meta name="keywords" content="rust, rustlang, rust-lang, spirv_std"><title>spirv_std - Rust</title><link rel="preload" as="font" type="font/woff2" crossorigin href="../SourceSerif4-Regular.ttf.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../FiraSans-Regular.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../FiraSans-Medium.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../SourceCodePro-Regular.ttf.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../SourceSerif4-Bold.ttf.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../SourceCodePro-Semibold.ttf.woff2"><link rel="stylesheet" type="text/css" href="../normalize.css"><link rel="stylesheet" type="text/css" href="../rustdoc.css" id="mainThemeStyle"><link rel="stylesheet" type="text/css" href="../ayu.css" disabled><link rel="stylesheet" type="text/css" href="../dark.css" disabled><link rel="stylesheet" type="text/css" href="../light.css" id="themeStyle"><script id="default-settings" ></script><script src="../storage.js"></script><script src="../crates.js"></script><script defer src="../main.js"></script>
    <noscript><link rel="stylesheet" href="../noscript.css"></noscript><link rel="alternate icon" type="image/png" href="../favicon-16x16.png"><link rel="alternate icon" type="image/png" href="../favicon-32x32.png"><link rel="icon" type="image/svg+xml" href="../favicon.svg"></head><body class="rustdoc mod crate"><!--[if lte IE 11]><div class="warning">This old browser is unsupported and will most likely display funky things.</div><![endif]--><nav class="mobile-topbar"><button class="sidebar-menu-toggle">&#9776;</button><a class="sidebar-logo" href="../spirv_std/index.html"><div class="logo-container"><img class="rust-logo" src="../rust-logo.svg" alt="logo"></div>
        </a><h2 class="location"></h2>
    </nav>
    <nav class="sidebar"><a class="sidebar-logo" href="../spirv_std/index.html"><div class="logo-container"><img class="rust-logo" src="../rust-logo.svg" alt="logo"></div>
        </a><h2 class="location"><a href="#">Crate spirv_std</a></h2><div class="sidebar-elems"><div class="block"><ul><li class="version">Version 0.4.0-alpha.14</li><li><a id="all-types" href="all.html">All Items</a></li></div></ul><section><div class="block"><ul><li><a href="#reexports">Re-exports</a></li><li><a href="#modules">Modules</a></li><li><a href="#macros">Macros</a></li><li><a href="#structs">Structs</a></li></ul></div></section><div id="sidebar-vars" data-name="spirv_std" data-ty="mod" data-relpath=""></div><script defer src="sidebar-items.js"></script></div></nav><main><div class="width-limiter"><div class="sub-container"><a class="sub-logo-container" href="../spirv_std/index.html"><img class="rust-logo" src="../rust-logo.svg" alt="logo"></a><nav class="sub"><form class="search-form"><div class="search-container"><span></span><input class="search-input" name="search" autocomplete="off" spellcheck="false" placeholder="Click or press ‘S’ to search, ‘?’ for more options…" type="search"><button type="button" id="help-button" title="help">?</button><div id="settings-menu" tabindex="-1">
                                <a href="../settings.html" title="settings"><img width="22" height="22" alt="Change settings" src="../wheel.svg"></a></div>
                        </div></form></nav></div><section id="main-content" class="content"><div class="main-heading">
    <h1 class="fqn"><span class="in-band">Crate <a class="mod" href="#">spirv_std</a><button id="copy-path" onclick="copy_path(this)" title="Copy item path to clipboard"><img src="../clipboard.svg" width="19" height="18" alt="Copy item path"></button></span></h1><span class="out-of-band"><a class="srclink" href="../src/spirv_std/lib.rs.html#1-151">source</a> · <a id="toggle-all-docs" href="javascript:void(0)" title="collapse all docs">[<span class="inner">&#x2212;</span>]</a></span></div><details class="rustdoc-toggle top-doc" open><summary class="hideme"><span>Expand description</span></summary><div class="docblock"><p>Core functions, traits, and more that make up a “standard library” for SPIR-V for use in
rust-gpu.</p>
</div></details><h2 id="reexports" class="small-section-header"><a href="#reexports">Re-exports</a></h2>
<div class="item-table"><div class="item-row"><div class="item-left"><code>pub extern crate <a class="mod" href="../spirv_std_macros/index.html" title="mod spirv_std_macros">spirv_std_macros</a> as macros;</code></div></div><div class="item-row"><div class="item-left import-item" id="reexport.ByteAddressableBuffer"><code>pub use byte_addressable_buffer::<a class="struct" href="byte_addressable_buffer/struct.ByteAddressableBuffer.html" title="struct spirv_std::byte_addressable_buffer::ByteAddressableBuffer">ByteAddressableBuffer</a>;</code></div><div class="item-right docblock-short"></div></div><div class="item-row"><div class="item-left import-item" id="reexport.num_traits"><code>pub use <a class="mod" href="../num_traits/index.html" title="mod num_traits">num_traits</a>;</code></div><div class="item-right docblock-short"></div></div><div class="item-row"><div class="item-left import-item" id="reexport.glam"><code>pub use <a class="mod" href="../glam/index.html" title="mod glam">glam</a>;</code></div><div class="item-right docblock-short"></div></div></div><h2 id="modules" class="small-section-header"><a href="#modules">Modules</a></h2>
<div class="item-table"><div class="item-row"><div class="item-left module-item"><a class="mod" href="arch/index.html" title="spirv_std::arch mod">arch</a></div><div class="item-right docblock-short"><p>SPIR-V Instrinics</p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="mod" href="byte_addressable_buffer/index.html" title="spirv_std::byte_addressable_buffer mod">byte_addressable_buffer</a></div><div class="item-right docblock-short"><p>Container for an untyped blob of data.</p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="mod" href="float/index.html" title="spirv_std::float mod">float</a></div><div class="item-right docblock-short"><p>Traits and helper functions related to floats.</p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="mod" href="image/index.html" title="spirv_std::image mod">image</a></div><div class="item-right docblock-short"><p>Image types</p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="mod" href="integer/index.html" title="spirv_std::integer mod">integer</a></div><div class="item-right docblock-short"><p>Traits related to integers.</p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="mod" href="memory/index.html" title="spirv_std::memory mod">memory</a></div><div class="item-right docblock-short"><p>Types for handling memory ordering constraints for concurrent memory access.</p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="mod" href="number/index.html" title="spirv_std::number mod">number</a></div><div class="item-right docblock-short"><p>Traits and helper functions related to numbers.</p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="mod" href="ray_tracing/index.html" title="spirv_std::ray_tracing mod">ray_tracing</a></div><div class="item-right docblock-short"><p>Ray-tracing data types</p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="mod" href="scalar/index.html" title="spirv_std::scalar mod">scalar</a></div><div class="item-right docblock-short"><p>Traits related to scalars.</p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="mod" href="vector/index.html" title="spirv_std::vector mod">vector</a></div><div class="item-right docblock-short"><p>Traits related to vectors.</p>
</div></div></div><h2 id="macros" class="small-section-header"><a href="#macros">Macros</a></h2>
<div class="item-table"><div class="item-row"><div class="item-left module-item"><a class="macro" href="macro.Image.html" title="spirv_std::Image macro">Image</a></div><div class="item-right docblock-short"><p>A macro for creating SPIR-V <code>OpTypeImage</code> types. Always produces a
<code>spirv_std::image::Image&lt;...&gt;</code> type.</p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="macro" href="macro.ray_query.html" title="spirv_std::ray_query macro">ray_query</a></div><div class="item-right docblock-short"><p>Constructs an uninitialized ray query variable. Using the syntax
<code>let (mut)? &lt;name&gt;</code>. Where <code>name</code> is the name of the ray query variable.</p>
</div></div></div><h2 id="structs" class="small-section-header"><a href="#structs">Structs</a></h2>
<div class="item-table"><div class="item-row"><div class="item-left module-item"><a class="struct" href="struct.RuntimeArray.html" title="spirv_std::RuntimeArray struct">RuntimeArray</a></div><div class="item-right docblock-short"><p>Dynamically-sized arrays in Rust carry around their length as the second half of a tuple.
Unfortunately, sometimes SPIR-V provides an unsized array with no way of obtaining its length.
Hence, this type represents something very similar to a slice, but with no way of knowing its
length.</p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="struct" href="struct.Sampler.html" title="spirv_std::Sampler struct">Sampler</a></div><div class="item-right docblock-short"><p>An opaque reference to settings that describe how to access, filter, or
sample an image.</p>
</div></div></div></section></div></main><div id="rustdoc-vars" data-root-path="../" data-current-crate="spirv_std" data-themes="ayu,dark,light" data-resource-suffix="" data-rustdoc-version="1.63.0-nightly (1fede1753 2022-05-28)" ></div>
</body></html>