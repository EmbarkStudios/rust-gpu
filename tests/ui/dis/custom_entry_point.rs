// build-pass
// compile-flags: -C llvm-args=--disassemble-globals

use spirv_std as _;

#[spirv(fragment(entry_point_name = "hello_world"))]
pub fn main() {}
