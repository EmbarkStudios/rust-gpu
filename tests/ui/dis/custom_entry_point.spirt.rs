// HACK(eddyb) duplicate of custom_entry_point.not_spirt.rs because only-/ignore- do not work with revisions.
// only-spirt

// build-pass
// compile-flags: -C llvm-args=--disassemble-globals
// normalize-stderr-test "OpCapability VulkanMemoryModel\n" -> ""
// normalize-stderr-test "OpExtension .SPV_KHR_vulkan_memory_model.\n" -> ""
// normalize-stderr-test "OpMemoryModel Logical Vulkan" -> "OpMemoryModel Logical Simple"

use spirv_std::spirv;

#[spirv(fragment(entry_point_name = "hello_world"))]
pub fn main() {}
