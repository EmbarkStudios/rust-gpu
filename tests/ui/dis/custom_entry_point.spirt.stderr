error: invalid character `'.'` in crate name: `custom_entry_point.spirt`

error: aborting due to previous error

